src/realm/util/network.cpp

```diff
@@ -1848,6 +1848,22 @@ void Service::Descriptor::accept(Descriptor& desc, StreamProtocol protocol, Endp
         }
 #endif
         new_sock_fd.reset(ret);
+
+#if REALM_PLATFORM_APPLE
+        int optval = 1;
+        ret = ::setsockopt(new_sock_fd, SOL_SOCKET, SO_NOSIGPIPE, &optval, sizeof optval);
+        if (REALM_UNLIKELY(ret == -1)) {
+            // setsockopt() reports EINVAL if the other side disconnected while
+            // the connection was waiting in the listen queue.
+            int err = errno;
+            if (err == EINVAL) {
+                continue;
```

Comment:
Opening and then closing a whole bunch of connections between when the server started listening and when the server's event loop started resulted in the server breaking, which is something that one of the audit tests wants to do. Just silently discarding the closed connections without ever marking them as valid avoids the problem.